# Hurtos en Bogotá durante el 2017:

In [285]:
import pandas as pd

In [286]:
#Hurto de celulares 2017 en toda colombia (primera componente)
df = pd.read_csv('/Users/jscs/Downloads/Hurto_a_personas_2017.csv',
            sep = ',', encoding = 'utf-8')
df.shape

(185105, 20)

In [287]:
#Datos de Bogota
df_bogota = df[df['Municipio'] == "BOGOTÁ D.C. (CT)"]
df_bogota.shape

(65482, 20)

In [288]:
#filtro transporte
df_bog_trans = df_bogota[df_bogota['Clase de sitio'].str.contains('TRANS')]
#filtro transporte 2
df_bog_trans = df_bog_trans[(df_bog_trans['Móvil Victima'] != "BICICLETA")&
                            (df_bog_trans['Móvil Victima'] != "PASAJERO VEHICULO")&
                            (df_bog_trans['Móvil Victima'] != "CONDUCTOR TAXI")&
                            (df_bog_trans['Móvil Victima'] != "PASAJERO TAXI")&
                            (df_bog_trans['Móvil Victima'] != "CONDUCTOR TAXI")&
                            (df_bog_trans['Móvil Victima'] != "CONDUCTOR MOTOCICLETA")&
                            (df_bog_trans['Móvil Victima'] != "PASAJERO BARCO")&
                            (df_bog_trans['Móvil Victima'] != "PASAJERO MOTOCICLETA")&
                            (df_bog_trans['Móvil Victima'] != "CONDUCTOR VEHICULO")&
                            (df_bog_trans['Móvil Victima'] != "TRIPULANTE AERONAVE")]
df_bog_trans.shape

(7200, 20)

In [289]:
p_nuevos_mas = df_bog_trans[df_bog_trans['Sexo'] == 'MASCULINO']
p_nuevos_fem = df_bog_trans[df_bog_trans['Sexo'] == 'FEMENINO']
trace1 = go.Histogram(
    x=p_nuevos_mas['Edad'].tolist(),
    opacity=0.75,
    name = 'Masculino'
)
trace2 = go.Histogram(
    x=p_nuevos_fem['Edad'].tolist(),
    opacity=0.75,
    name = 'Femenino'
)

data = [trace1, trace2]
layout = go.Layout(barmode='overlay',
    title='Histograma de la variable edad con estratificación de género',
)
go.FigureWidget(data=data, layout=layout)

FigureWidget({
    'data': [{'name': 'Masculino',
              'opacity': 0.75,
              'type': 'histog…

In [290]:
#filtro edad
df_bog_trans = df_bog_trans[(df_bog_trans['Edad'] > 17) & (df_bog_trans['Edad'] < 33)]
df_bog_trans.shape

(3950, 20)

In [291]:
df_bog_sitio = pd.DataFrame(df_bog_trans['Clase de sitio'].value_counts())
total = df_bog_trans['Clase de sitio'].sum()

df_bog_sitio = df_bog_sitio[:25]
df_bog_sitio['Porcentaje'] =  round((df_bog_sitio['Clase de sitio']*100)/3950,1)
df_bog_sitio

Clase de sitio  Porcentaje
BUS TRANSMILENIO                                    715        18.1
TRANSPORTE PÚBLICO                                  590        14.9
ALIMENTADOR BUS TRANSMILENIO                        457        11.6
ESTACION TRANSMILENIO (INICIAL)                     140         3.5
ESTACION TRANSMILENIO RICAURTE (CLL 13)              62         1.6
ESTACION TRANSMILENIO PORTAL DE LA 80                55         1.4
ESTACION TRANSMILENIO PORTAL NORTE                   54         1.4
ALIMENTADOR BUS TRANSMETRO                           54         1.4
ESTACION TRANSMILENIO AV. JIMENEZ                    51         1.3
ESTACION TRANSMILENIO CALLE 100                      50         1.3
ESTACION TRANSMILENIO ALCALA                         49         1.2
ESTACION TRANSMILENIO MARLY                          47         1.2
ESTACION TRANSMILENIO UNIVERSIDADES                  46         1.2
ESTACION TRANSMILENIO PORTAL AMERICAS                46         1.2
ESTACION TRANSMILENIO BANDERAS                       44         1.1
ESTACION TRANSMILENIO PORTAL DEL SUR                 41         1.0
TERMINAL DE TRANSPORTES                              41         1.0
ESTACION TRANSMILENIO PORTAL DE SUBA                 40         1.0
ESTACION TRANSMETRO                                  39         1.0
ESTACION TRANSMILENIO RICAURTE (CRA 30)              39         1.0
ESTACION TRANSMILENIO CALLE 45                       38         1.0
ESTACION TRANSMILENIO AGUAS                          36         0.9
ESTACION TRANSMILENIO CALLE 63                       34         0.9
ESTACION TRANSMILENIO CALLE 72                       32         0.8
ESTACION TRANSMILENIO BIB. EL TINTAL                 30         0.8

In [292]:
df_edad = pd.DataFrame(df_bog_trans['Edad'].value_counts())
total = df_edad['Edad'].sum()
df_edad['Porcentaje'] =  round(df_edad['Edad']*100/total,1)
df_edad_15 = df_edad.head(15)
df_edad_15

Edad  Porcentaje
21.0   333         8.4
22.0   302         7.6
24.0   293         7.4
25.0   292         7.4
23.0   290         7.3
20.0   282         7.1
26.0   260         6.6
27.0   257         6.5
28.0   255         6.5
19.0   254         6.4
18.0   245         6.2
29.0   235         5.9
30.0   223         5.6
31.0   216         5.5
32.0   213         5.4

In [293]:
df_sexo = pd.DataFrame(df_bog_trans['Sexo'].value_counts())
total = df_sexo['Sexo'].sum()
df_sexo['Porcentaje'] =  round(df_sexo['Sexo']*100/total,1)
df_sexo

Sexo  Porcentaje
MASCULINO  2023        51.2
FEMENINO   1927        48.8

In [294]:
#Lugares más frecuentes
df_car1 = pd.DataFrame(df_bog_trans['Barrio'].value_counts())
df_car1['Porcentaje'] =  round(df_car1['Barrio']*100/total,1)

df_car1.head(10)

Barrio  Porcentaje
RICAURTE E-14             123         3.1
CHAPINERO CENTRAL E-2      91         2.3
TEUSAQUILLO E-13           67         1.7
MARLY E-2                  62         1.6
PATIO BONITO I E-8         61         1.5
LAS NIEVES E-3             48         1.2
CENTRO E-3                 41         1.0
LAS AGUAS E-3              40         1.0
BANDERAS E-8               39         1.0
SAN VICTORINO E-3          37         0.9

In [295]:
df_car2 = pd.DataFrame(df_bog_trans['Arma empleada'].value_counts())
total = df_car2['Arma empleada'].sum()
df_car2['Porcentaje'] =  round(df_car2['Arma empleada']*100/total,1)
df_car2 = df_car2.reset_index()
df_car2

index  Arma empleada  Porcentaje
0          SIN EMPLEO DE ARMAS           2785        70.5
1  ARMA BLANCA / CORTOPUNZANTE            964        24.4
2                ARMA DE FUEGO            106         2.7
3                 CONTUNDENTES             81         2.1
4                 ESCOPOLAMINA              9         0.2
5                    CORTANTES              3         0.1
6                        PERRO              1         0.0
7                 NO REPORTADO              1         0.0

In [296]:
labels = df_car2['index'].tolist()
values = df_car2['Arma empleada'].tolist()
trace = go.Pie(labels=labels, values=values,showlegend=True,hole=0,textposition='inside')
data = [trace]
layout = go.Layout(title='Distribución del arma empleada en los hurtos de celulares en el sistema de transporte de Bogotá')
go.FigureWidget(data=data, layout=layout)

FigureWidget({
    'data': [{'hole': 0,
              'labels': [SIN EMPLEO DE ARMAS, ARMA BLANCA / CORTOPUNZA…

In [297]:
df_car2 = pd.DataFrame(df_bog_trans['Escolaridad'].value_counts())
total = df_car2['Escolaridad'].sum()
df_car2['Porcentaje'] =  round(df_car2['Escolaridad']*100/total,1)
df_car2

Escolaridad  Porcentaje
SUPERIOR             2091        52.9
SECUNDARIA            984        24.9
TECNICO               419        10.6
TECNOLOGO             346         8.8
PRIMARIA               81         2.1
NO REPORTADO           28         0.7
-                       1         0.0

In [298]:
df_car2['Escolaridad'].sum()

3950

In [299]:
#pacientes nuevos por día
df_dias = pd.DataFrame(df_bog_trans['Fecha'].value_counts()) #Contar valores de cada ciudad
df_dias.reset_index(level=0, inplace=True) #organizar indexado
df_dias.columns = ['Fecha', 'Departamento'] #nombrar columnas
df_dias = df_dias.sort_values(by=['Fecha']) #orden de meses
#df_dias.T
#dejar solo gráfica

In [300]:
import plotly.graph_objs as go

trace = go.Scatter(
    x = df_dias["Fecha"].tolist(),
    y = df_dias["Departamento"].tolist()
)
data = [trace]
layout = dict(title = 'Hurtos de celular en el sistema de transporte de Bogotá',
              xaxis = dict(title = 'Fechas'),
              yaxis = dict(title = 'Hurtos'),
              )
go.FigureWidget(data=data, layout=layout)

FigureWidget({
    'data': [{'type': 'scatter',
              'uid': '6e8c1aeb-8540-4c0c-8ff5-10bee0594044',
 …

El 35% de los hurtos de celulares en Colombia suceden en Bogotá. El 11% de hurtos de celulares en Bogotá pasa en el sitema de transporte (enfocado en el Transmilenio). Cerca del 55% de las personas victimas de hurto en Bogotá en el Sistema de transporte están entre 18 años y 32 años. 